pyproject.toml

```diff
@@ -1,3 +1,156 @@
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "e3sm_diags"
+dynamic = ["version"]
+description = "E3SM Diagnostics"
+authors = [
+  { name = "Chengzhu (Jill) Zhang", email = "zhang40@llnl.gov" },
+  { name = "Tom Vo" },
+  { name = "Ryan Forsyth" },
+  { name = "Chris Golaz" },
+  { name = "Zeshawn Shaheen" },
+]
+license = { text = "BSD 3-Clause" }
+readme = "README.md"
+requires-python = ">=3.9"
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: BSD 3-Clause License",
+  "Natural Language :: English",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+]
+dependencies = [
+  "beautifulsoup4",
+  "cartopy >=0.17.0",
+  # This package is not available on PyPI.
+  # "cartopy_offlinedata",
+  "cf-units",
+  "dask",
+  "esmpy >=8.4.0",
+  "lxml",
+  "mache >=0.15.0",
+  # This package is not available on PyPI.
+  # "matplotlib-base",
+  "netcdf4",
+  "output_viewer >=1.3.0",
+  "numpy >=2.0.0,<3.0.0",
+  "shapely >=2.0.0,<3.0.0",
+  "xarray >=2024.03.0",
+  "xcdat >=0.7.3,<1.0.0",
+  "xesmf >=0.8.7",
+  "xskillscore >=0.0.20",
+]
+
+[project.optional-dependencies]
+test = ["pytest", "pytest-cov"]
+docs = ["sphinx", "sphinx_rtd_theme", "sphinx-multiversion"]
+dev = [
+  "black",
+  "flake8",
+  "flake8-isort",
+  "isort",
+  "mypy",
+  "pre-commit",
+  "types-PyYAML",
+]
+
+[project.urls]
+Documentation = "https://docs.e3sm.org/e3sm_diags/_build/html/main/index.html"
+"Issue Tracker" = "https://github.com/E3SM-Project/e3sm_diags/issues"
+
+[project.scripts]
+e3sm_diags = "e3sm_diags.e3sm_diags_driver:main"
+e3sm_diags_vars = "e3sm_diags.e3sm_diags_vars:main"
+
+[tool.setuptools.packages.find]
+include = ["e3sm_diags", "e3sm_diags.*"]
+
+[tool.setuptools.dynamic]
+version = { attr = "e3sm_diags.__version__" }
+
+[tool.setuptools.data-files]
```

Comment:
I agree, there are a ton of files being moved around in the old `setup.py` file that I didn't take the time to understand. This can be done a separate PR. Thanks.

Reply:
Related PR: https://github.com/E3SM-Project/e3sm_diags/issues/897